src/test/scala/sigmastate/TestingInterpreterSpecification.scala

```diff
@@ -195,10 +195,10 @@ class TestingInterpreterSpecification extends SigmaTestingCommons {
   }
 
   property("failed numeric downcast (overflow)") {
-    an[ArithmeticException] should be thrownBy testEval("Array(999)(0).toByte > 0")
-    an[ArithmeticException] should be thrownBy testEval("Array(999)(0).toShort.toByte > 0")
-    an[ArithmeticException] should be thrownBy testEval(s"Array(${Int.MaxValue})(0).toShort > 0")
-    an[ArithmeticException] should be thrownBy testEval(s"Array(${Long.MaxValue}L)(0).toInt > 0")
+    an[IR.StagingException] should be thrownBy testEval("Array(999)(0).toByte > 0")
+    an[IR.StagingException] should be thrownBy testEval("Array(999)(0).toShort.toByte > 0")
+    an[IR.StagingException] should be thrownBy testEval(s"Array(${Int.MaxValue})(0).toShort > 0")
+    an[IR.StagingException] should be thrownBy testEval(s"Array(${Long.MaxValue}L)(0).toInt > 0")
```

Comment:
@greenhat we actually need to check that StagingException has caused by expected exception.

Reply:
@ergomorphic Done.